middle_end/flambda.mli

```diff
@@ -553,13 +554,28 @@ val create_function_declaration
   -> inline:Lambda.inline_attribute
   -> specialise:Lambda.specialise_attribute
   -> is_a_functor:bool
+  -> closure_origin:Closure_origin.t
   -> function_declaration
 
 (** Create a set of function declarations given the individual declarations. *)
 val create_function_declarations
    : funs:function_declaration Variable.Map.t
   -> function_declarations
 
+(** Change only the code of a function declaration. *)
+val update_body_of_function_declaration
+   : function_declaration
+  -> body:expr
+  -> function_declaration
```

Comment:
I'm not completely certain I like that kind of functions. Usualy when adding a new field in function_declaration, it is a good thing to think about every place where it needs to be updated. But I understand that it makes some changes far less tedious.